File: src/main/java/com/rs/java/game/area/shapes/Rectangle.java
package com.rs.java.game.area.shapes;

import com.rs.java.game.WorldTile;
import com.rs.java.game.area.Shape;

public class Rectangle extends Shape {

	private final WorldTile sw;
	private final WorldTile ne;
	private final String debugLabel;

	public Rectangle(WorldTile southWest, WorldTile northEast) {
		this(southWest, northEast, null);
	}

	public Rectangle(WorldTile southWest, WorldTile northEast, String debugLabel) {
		this.sw = southWest;
		this.ne = northEast;
		this.debugLabel = debugLabel;

		areas(new WorldTile[]{ ne, sw }).type(ShapeType.RECTANGLE);

	}

	@Override
	public boolean inside(WorldTile location) {
		boolean inPlane = location.getPlane() == sw.getPlane() && location.getPlane() == ne.getPlane();
		boolean inX = location.getX() >= sw.getX() && location.getX() <= ne.getX();
		boolean inY = location.getY() >= sw.getY() && location.getY() <= ne.getY();

        return inPlane && inX && inY;
	}


	public WorldTile getSouthWest() { return sw; }
	public WorldTile getNorthEast() { return ne; }
}

File: src/main/java/com/rs/kotlin/game/world/area/shape/Rectangle.kt
package com.rs.kotlin.game.world.area.shape

import com.rs.java.game.WorldTile
import com.rs.kotlin.game.world.area.Shape

class Rectangle(
    private val sw: WorldTile,
    private val ne: WorldTile,
    private val debugLabel: String? = null
) : Shape() {

    init {
        areas(arrayOf(ne, sw)).type(ShapeType.RECTANGLE)
    }

    override fun inside(location: WorldTile): Boolean {
        val inPlane = location.plane == sw.plane && location.plane == ne.plane
        val inX = location.x in sw.x..ne.x
        val inY = location.y in sw.y..ne.y
        return inPlane && inX && inY
    }

    fun getSouthWest(): WorldTile = sw
    fun getNorthEast(): WorldTile = ne
}
